src/main/java/org/folio/services/piece/PieceService.java

```diff
@@ -54,17 +56,22 @@ public Future<List<Piece>> getPiecesByPoLineId(String poLineId, DBClient client)
 
   private Future<Tx<PoLine>> updatePieces(Tx<PoLine> poLineTx, List<Piece> pieces, DBClient client) {
     Promise<Tx<PoLine>> promise = Promise.promise();
-    String query = buildUpdatePieceBatchQuery(pieces, client.getTenantId());
 
-    client.getPgClient().execute(poLineTx.getConnection(), query, reply -> {
-      if (reply.failed()) {
-        logger.error("Update Pieces failed : {}", reply);
-        httpHandleFailure(promise, reply);
-      } else {
-        logger.info("Pieces was successfully updated");
-        promise.complete(poLineTx);
-      }
-    });
+    if(CollectionUtils.isNotEmpty(pieces)) {
+      String query = buildUpdatePieceBatchQuery(pieces, client.getTenantId());
+      client.getPgClient().execute(poLineTx.getConnection(), query, reply -> {
+        if (reply.failed()) {
+          logger.error("Update Pieces failed : {}", reply);
+          httpHandleFailure(promise, reply);
+        } else {
+          logger.info("Pieces was successfully updated");
+          promise.complete(poLineTx);
+        }
+      });
+    } else {
+      logger.info(PIECE_NOT_UPDATED, poLineTx.getEntity().getId());
```

Comment:
I think we can concretize this message, in our case pieces not updated because nothing to update, so we can write log like this:
logger.info("Pieces for po line not presented, skipping the update", poLineTx.getEntity().getId());

Reply:
Thanks. Updated.